core/src/main/java/org/apache/accumulo/core/fate/accumulo/AccumuloStore.java

```diff
@@ -0,0 +1,288 @@
+    try (Scanner scanner = context.createScanner(tableName, Authorizations.EMPTY)) {
+      return func.apply(scanner);
+    } catch (TableNotFoundException e) {
+      throw new IllegalStateException(tableName + " not found!", e);
+    }
+  }
+
+  private class FateTxStoreImpl extends AbstractFateTxStoreImpl<T> {
+
+    private FateTxStoreImpl(long tid, boolean isReserved) {
+      super(tid, isReserved);
+    }
+
+    @Override
+    public Repo<T> top() {
+      verifyReserved(false);
+
+      return scanTx(scanner -> {
+        scanner.setRange(getRow(tid));
+        scanner.fetchColumnFamily(RepoColumnFamily.NAME);
+        return StreamSupport.stream(scanner.spliterator(), false).sorted(repoComparator).map(e -> {
+          @SuppressWarnings("unchecked")
+          var repo = (Repo<T>) deserialize(e.getValue().get());
+          return repo;
+        }).findFirst().orElse(null);
+      });
+    }
+
+    @Override
+    public List<ReadOnlyRepo<T>> getStack() {
+      verifyReserved(false);
+
+      return scanTx(scanner -> {
+        scanner.setRange(getRow(tid));
+        scanner.fetchColumnFamily(RepoColumnFamily.NAME);
+        return StreamSupport.stream(scanner.spliterator(), false).sorted(repoComparator).map(e -> {
+          @SuppressWarnings("unchecked")
+          var repo = (ReadOnlyRepo<T>) deserialize(e.getValue().get());
+          return repo;
+        }).collect(Collectors.toList());
+      });
+    }
+
+    @Override
+    public Serializable getTransactionInfo(TxInfo txInfo) {
+      verifyReserved(false);
+
+      try (Scanner scanner = context.createScanner(tableName, Authorizations.EMPTY)) {
+        scanner.setRange(new Range("tx_" + FastFormat.toHexString(tid)));
+
+        final ColumnFQ cq;
+        switch (txInfo) {
+          case TX_NAME:
+            cq = TxInfoColumnFamily.TX_NAME_COLUMN;
+            break;
+          case AUTO_CLEAN:
+            cq = TxInfoColumnFamily.AUTO_CLEAN_COLUMN;
+            break;
+          case EXCEPTION:
+            cq = TxInfoColumnFamily.EXCEPTION_COLUMN;
+            break;
+          case RETURN_VALUE:
+            cq = TxInfoColumnFamily.RETURN_VALUE_COLUMN;
+            break;
+          default:
+            throw new IllegalArgumentException("Unexpected TxInfo type " + txInfo);
+        }
+        scanner.fetchColumn(cq.getColumnFamily(), cq.getColumnQualifier());
+
+        return StreamSupport.stream(scanner.spliterator(), false)
+            .map(e -> deserializeTxInfo(txInfo, e.getValue().get())).findFirst().orElse(null);
+      } catch (TableNotFoundException e) {
+        throw new IllegalStateException(tableName + " not found!", e);
+      }
+    }
+
+    @Override
+    public long timeCreated() {
+      verifyReserved(false);
+
+      return scanTx(scanner -> {
+        scanner.setRange(getRow(tid));
+        scanner.fetchColumn(TxColumnFamily.CREATE_TIME_COLUMN.getColumnFamily(),
+            TxColumnFamily.CREATE_TIME_COLUMN.getColumnQualifier());
```

Comment:
Can do the following to shorten the code.

```suggestion
        TxColumnFamily.CREATE_TIME_COLUMN.fetch(scanner);
```